lib/ansible/modules/lineinfile.py

```diff
@@ -347,9 +400,9 @@ def present(module, dest, regexp, line, insertafter, insertbefore, create,
         if not b_new_line.endswith(b_linesep):
             b_new_line += b_linesep
 
-        # If no regexp was given and no line match is found anywhere in the file,
+        # If no regexp or literal was given and no line match is found anywhere in the file,
         # insert the line appropriately if using insertbefore or insertafter
-        if regexp is None and match is None and not exact_line_match:
+        if regexp is None and literal is None and match is None and not exact_line_match:
```

Comment:
This might read a bit easier:
```suggestion
        if (regexp, literal, match) == (None, None, None) and not exact_line_match:
```